check_msid_status.py

```diff
@@ -13,102 +13,82 @@
 #-------------------------------------------------------------------------------
 #-- check_status: check status of msid                                        --
 #-------------------------------------------------------------------------------
-import traceback
 
 def check_status(msid, val, ldict, vdict):
     """
     check status of msid
     input:  msid    --- msid
             val     --- current value of msid
-            ldict   --- dictionary of limits for nemeric entry
+            ldict   --- dictionary of limits for numeric entry
             vdict   --- dictionary of msid <---> current value
     output: status  --- current status of msid
     """
 #
-#--- the value is nemeric
+#--- If value is numeric, then convert
 #
     try:
         val = float(val)
-        try:
-            status = check_status_neumeric(msid, val, ldict)
-            return status
-        except:
-            return 'GREEN'
+    except:
+        pass
 #
-#--- the value is letters
+#--- General Violation
 #
-    except:
-        if val in ['NaN', 'nan', '']:
-            return 'CAUTION'
-
-        elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
-                    'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
-                    'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR',\
-                    '2SHLDART', 'PLINE03T', 'PLINE04T', 'AACCCDPT', '3LDRTNO']:
-            try:
-                status = check_status_letter(msid, val, vdict)
-                return status
-            except KeyError:
-                return 'GREEN'
-            except:
-                traceback.print_exc()
-                return 'GREEN'
-        else:
-            if val in ['ERR', 'FALT', 'FAIL']:
-                return "CAUTION"
-            else:
-                return 'GREEN'
+    if val in ['ERR', 'FALT', 'FAIL', 'NaN', 'nan', '']:
+        return 'CAUTION'
+#
+#--- Edge Case Violation
+#
+    elif msid in ['AOCONLAW', 'AOCPESTL', '4OBAVTMF', '4OBTOORF', 'COSCS128S',\
+                  'COSCS129S','COSCS130S', 'COSCS131S','COSCS132S','COSCS133S',\
+                  'COSCS107S','CORADMEN', 'CCSDSTMF', 'ACAFCT', 'AOFSTAR', '2SHLDART']:
```

Comment:
The MSIDs 'PLINE03T', 'PLINE04T', 'AACCCDPT', '3LDRTNO' are all numerically valued MSIDs which have limits stored in the CSH_limit_table.json file. Therefore they do not need to be part of the edge case function.

Reply:
It seems that only 4OBAVTMF and 4OBTOORF may result in val == 'FALT' and I think this case is taken into account in check_status_edge_case(), lines 111-114, so likely 'FALT' can be removed from the list in line 36.

For which of these msids may give val = 'ERR' or val == 'FAIL'?

Reply:
> OK.
> 
> I noticed that https://github.com/chandra-mta/CSH/blob/fetch_blobs/check_msid_status.py contains AOACFCTC (which does not exist) instead of ACAFCT (which also does not exist) in line 43. Why? The most recent code version on the fetch_blobs branch and this PR should match. Do ACAFCT and AOACFCTC belong to MTA pseudomsids?

The discrepancy between these two names on Github is due to the link I sent of selected github commits for the second round of review. I original sent the link which consisted of the 13 most recent commits, then after some time I discovered this misnamed psuedo-msids and made a new commit to fix the issue. Both are MTA psuedo-msids for the ACA image, yet the status check developed in check_msid_status.py was inconsistent with the rest of the script. This has now been corrected.

Reply:
> It seems that only 4OBAVTMF and 4OBTOORF may result in val == 'FALT' and I think this case is taken into account in check_status_edge_case(), lines 111-114, so likely 'FALT' can be removed from the list in line 36.
> 
> For which of these msids may give val = 'ERR' or val == 'FAIL'?

In investigating the edge cases, none of them give the value of "ERR" or "FAIL" when investigating their values during the time period 2022:001 to 2024:001. Therefore removing FALT, 'NaN','nan', and '' from the list of general violations.

Upon investigating a set of values in the ccdm section, I believe I found a significant enough set of possible ERR or FAIL values msids such that they warrant keeping in the list in line 36.

Possible ERR values
["CIUACAPA", "CIUACBPA", "CIUBCAPA", "CIUBCBPA", "CPCPEPR", "CPCPERD", "CSIPEPR", "CSIPERD", "CTSPEPR", "CTSPERD"]

Possible FAIL values
["CM1CKBP1", "CM1CKBP2", "CM1CKBR1", "CM1CKBR2", "CM2CKBP1", "CM2CKBP2", "CM2CKBR1", "CM2CKBR2", "CM3CKBP1", "CM3CKBP2", "CM3CKBR1", "CM3CKBR2"]

Note that these two lists above are not exhaustive of all checked MSIDS which could have ERR, or FAIL values, but considering how many of them there are already, I believe it more clear to check against a possible "ERR"or "FAIL" value directly rather than implement the above selection into the edge case checks.